pkg/services/pluginsintegration/keyretriever/dynamic/dynamic_retriever.go

```diff
@@ -134,10 +135,20 @@ func (kr *KeyRetriever) downloadKeys(ctx context.Context) error {
 		}
 	}()
 
-	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
```

Comment:
Should we be closing the response body once we're done with it?

Reply:
yep, it's there already (line 132)